### PR TITLE
Device info endpoint

### PR DESCRIPTION
**Pull request by rtibbles**
# Checklist

- [x] PR has the correct target milestone when it's merged
- [ ] Automated test coverage is satisfactory
- [x] PR has been fully tested manually
- [ ] Documentation is updated as necessary
- [ ] External dependency files are updated (`yarn` and `pip`)
- [ ] If internal dependency are updated, link to diff is included
- [ ] Screenshots of any front-end changes are in the PR description
- [ ] CHANGELOG.rst is updated for high-level changes
- [ ] You've added yourself to AUTHORS.rst if you're not there

# Details

### Summary

Implements an API endpoint to provide data for #1910.

Returns: version, URLs, DB Path (when available), Device Name, Morango Device ID, OS, Free Space in Content Drive, Server Date and Time, Server Timezone.

Also adds a function on the frontend to retrieve from the endpoint, and do some initial data validation and cleaning on the clientside.

### CODE REVIEW

File: kolibri/core/device/api.py [hunk 1]
```diff
@@ -24,3 +33,46 @@ def create(self, request, *args, **kwargs):
...
+
+        info['device_name'] = platform.node()
+        info['device_id'] = InstanceIDModel.get_or_create_current_instance()[0].id
+        info['os'] = platform.platform()
+
+        # Will either return a named tuple or None if the drive cannot be found
+        drive = get_drive_data(get_content_storage_folder_path())
+
+        if drive:
+            info['content_storage_free_space'] = drive.freespace
```
Comment (rtibbles) [inline kolibri/core/device/api.py]: Can update this when #2482 has been merged, which has a utility specifically for this!

**Comment by rtibbles** [comment]
Updated to use @lyw07's free disk space utility.

File: kolibri/core/device/api.py [hunk 2]
```diff
@@ -38,3 +46,34 @@ def list(self, request):
...
+        if not urls:
+            # Will not return anything when running the debug server, so at least return the current URL
+            urls = [request.build_absolute_uri('/')]
+        info['urls'] = urls
+
+        if settings.DATABASES['default']['ENGINE'].endswith('sqlite3'):
+            # If any other database backend, will not be file backed, so no database path to return
+            info['database_path'] = settings.DATABASES['default']['NAME']
+
+        info['device_name'] = platform.node()
```
Comment (ralphiee22) [inline kolibri/core/device/api.py]: you _could_ get this from InstanceIDModel.hostname

File: kolibri/core/device/api.py [hunk 3]
```diff
@@ -38,3 +46,34 @@ def list(self, request):
...
+        info['device_id'] = InstanceIDModel.get_or_create_current_instance()[0].id
+        info['os'] = platform.platform()
```
Comment (ralphiee22) [inline kolibri/core/device/api.py]: as well you _could_ get this from InstanceIDModel.platform

File: kolibri/core/device/test/test_api.py [hunk 4]
```diff
@@ -204,6 +206,16 @@ def test_urls(self):
             # Make sure each url is a valid link
             self.assertTrue(url.startswith('http://'))
 
+    @patch('kolibri.core.device.api.get_urls', return_value=(1, ['http://127.0.0.1:8000', 'http://kolibri.com']))
+    def test_no_localhost_urls_when_others_available(self, get_urls_mock):
+        response = self.client.get(reverse('deviceinfo'), format="json")
+        self.assertEqual(len(response.data['urls']), 1)
```
Comment (ralphiee22) [inline kolibri/core/device/test/test_api.py]: how do we know that `http://127.0.0.1:8000` was not returned?

Comment (rtibbles) [reply kolibri/core/device/test/test_api.py]: True.

**Pull request merged**